src/openrct2/Context.cpp

```diff
@@ -103,8 +104,12 @@ namespace OpenRCT2
         Networking::Http::Http _http;
 #endif
 
-        // Game states
+        // Screens.
         std::unique_ptr<TitleScreen> _titleScreen;
+        std::unique_ptr<GameScreen> _gameScreen;
+        IScreenState* _activeScreen = nullptr;
```

Comment:
What about making the `std::unique_ptr` into `shared` and having `_activeScreen` be a `weak_ptr`, for safety?

Reply:
Since you really only want one instance of each screen I would think `std::unique_ptr` would be the appropriate choice here. Everywhere the raw pointer is used it is checked to see if it is null so I don't think it can cause a problem. Ideally you would want to make `_activeScreen` a `std::observer_ptr` to emphasize that it is non-owning but I don't think observer pointers have been officially adopted to the standard yet so a raw pointer is probably the best you can really do for now.

Reply:
I would agree if it wouldn't be clear when the memory gets released, the context drives the lifetime, but yeah I'm not a fan of pointers. I wan't to cleanup the whole initialization at some point its a mess atm.

Reply:
Alright

Reply:
Not a fan of pointers at all??? or just not a fan of raw pointers? It's hard to imagine being a C++ programmer and not liking pointers at all lol

Reply:
Yes don't use pointers if they can be avoided, and in this case they probably can be avoided by making them basic members, so when the context would be destroyed so is everything with it. If you have some function called Initialize in a class you are breaking the core concept of C++ that wants objects to be usable after construction.

Reply:
I guess I can agree with that, although wouldn't any `unique_ptr` class members automatically be destroyed along with the class too?

Reply:
Yes they would be destroyed automatically but they are still dynamic allocations which in this case is not required at all if I would just have them as basic members, ex.: "GameScreen _gameScreen" and pass the GameState object during construction and then everything is nicely in-place in the Context class.

Reply:
Ah I see what you mean. Sorry if I ask a lot of questions, its refreshing to talk to people who actually deeply understand C++ haha. I can finally get answers to some questions people I work with don't have clear answers to.